src/Angor/Client/Pages/Signatures.razor

```diff
@@ -408,103 +423,118 @@
         }
     }
 
-    private async Task<OperationResult> PerformSignatureApproval(SignatureRequest signature, WalletWords words)
+    private async Task ApproveSignature(SignatureRequest signature)
     {
+        signaturesRequestsApproving.Add(signature, string.Empty);
+        StateHasChanged();
+
         try
         {
-            if (words == null)
-            {
-                return new OperationResult { Success = false, Message = "Password retrieval failed." };
-            }
-
-            var key = DerivationOperations.DeriveFounderRecoveryPrivateKey(words, FounderProject.ProjectIndex);
-            var signatureInfo = signProject(signature.TransactionHex, FounderProject.ProjectInfo, Encoders.Hex.EncodeData(key.ToBytes()));
-
-            var sigJson = serializer.Serialize(signatureInfo);
-
-            var nostrPrivateKey = await DerivationOperations.DeriveProjectNostrPrivateKeyAsync(words, FounderProject.ProjectIndex);
-            var nostrPrivateKeyHex = Encoders.Hex.EncodeData(nostrPrivateKey.ToBytes());
-
-            var encryptedContent = await javascriptNostrToolsModule.InvokeAsync<string>(
-                "encryptNostr",
-                nostrPrivateKeyHex,
-                signature.investorPubKey,
-                sigJson);
-
-            FounderProject.LastRequestForSignaturesTime = SignService.SendSignaturesToInvestor(encryptedContent, nostrPrivateKeyHex, signature.investorPubKey, signature.EventId);
-            Storage.UpdateFounderProject(FounderProject);
+            var words = await passwordComponent.GetWalletAsync();
 
-            signaturesRequests.Single(_ => _.investorPubKey == signature.investorPubKey && _.TimeApproved is null)
-                .TimeApproved = FounderProject.LastRequestForSignaturesTime;
+            var operationResult = await PerformSignatureApproval(signature, words);
 
-            return new OperationResult { Success = true };
+            if (!operationResult.Success)
+            {
+                notificationComponent.ShowErrorMessage(operationResult.Message);
+            }
         }
-        catch (Exception ex)
+        catch (Exception e)
         {
-            return new OperationResult { Success = false, Message = $"An error occurred: {ex.Message}" };
+            Logger.LogError(e, e.Message);
+            notificationComponent.ShowErrorMessage(e.Message);
+        }
+        finally
+        {
+            signaturesRequestsApproving.Remove(signature);
         }
+
+        StateHasChanged();
```

Comment:
makes sense that we want to ensure the state change only happens if an exception wasn't thrown, nice catch!

Reply:
I believe there is a state change also inside the error  message component